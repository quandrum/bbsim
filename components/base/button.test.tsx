import { render, screen, fireEvent } from '@testing-library/react';
import Button from './button';

describe('Button', () => {
  const DEFAULT_PROPS = {
    onClick: jest.fn(),
    text: 'test',
  };

  beforeEach(DEFAULT_PROPS.onClick.mockReset);

  it('renders', () => {
    const { container } = render(<Button {...DEFAULT_PROPS} />);
    expect(container.innerHTML).toMatchSnapshot();
  });
  it('calls onClick when clicked', () => {
    render(<Button {...DEFAULT_PROPS} />);
    fireEvent.click(screen.getByText(DEFAULT_PROPS.text));
    expect(DEFAULT_PROPS.onClick).toHaveBeenCalled();
  });
});
