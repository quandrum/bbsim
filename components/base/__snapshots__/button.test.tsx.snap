// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `"<button type=\\"submit\\" class=\\"py-4 mt-4 mb-20 font-sans text-xl leading-tight text-center text-white bg-yellow-600 rounded px-17 md:px-12 md:py-4 md:text-base\\">test</button>"`;
